spec/spec_helper/index.rb

```diff
@@ -78,10 +78,24 @@ def sort_dependencies(dependencies, activated, conflicts)
         name = name_for(dependency)
         [
           activated.vertex_named(name).payload ? 0 : 1,
+          amount_constrained(dependency),
           conflicts[name] ? 0 : 1,
           activated.vertex_named(name).payload ? 0 : search_for(dependency).count,
         ]
       end
     end
+
+    def amount_constrained(dependency)
+      @amount_constrained ||= {}
+      @amount_constrained[dependency.name] ||= begin
+        all = specs[dependency.name].size
+        if all <= 1
+          all #- 1_000_000
```

Comment:
this improves performance in Bundler but given its not in `master` yet, I felt it wasn't right to commit it here